src/ast/data_type.rs

```diff
@@ -300,7 +311,7 @@ pub enum DataType {
     /// [clickhouse]: https://clickhouse.com/docs/en/sql-reference/data-types/nested-data-structures/nested
     Nested(Vec<ColumnDef>),
     /// Enums
-    Enum(Vec<String>),
+    Enum(Vec<EnumValue>, Option<i64>),
```

Comment:
```suggestion
    Enum(Vec<EnumValue>, Option<i64>),
```
is a negative size allowed (wondering if this should be u64 otherwise)?